com.hazelcast.client.impl.spi.impl.discovery.ClientAutoDetectionDiscoveryTest.defaultDiscovery [HZ-753]

_5.0.z_ (commit 008f2fdfb87d00bed17092825b576ecb3df6665c)

Failed on Zulu JDK 11: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-ZuluJDK11/3/testReport/com.hazelcast.client.impl.spi.impl.discovery/ClientAutoDetectionDiscoveryTest/defaultDiscovery/

Stacktrace:
```
java.lang.AssertionError: Cluster size is not correct. Expected: 2, actual: 1, instance index: 0
	at org.junit.Assert.fail(Assert.java:89)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:1047)
	at com.hazelcast.test.HazelcastTestSupport$11.run(HazelcastTestSupport.java:1075)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1260)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1277)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1072)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1060)
	at com.hazelcast.client.impl.spi.impl.discovery.ClientAutoDetectionDiscoveryTest.defaultDiscovery(ClientAutoDetectionDiscoveryTest.java:57)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Standard output:
```
22:50:39,887  INFO |defaultDiscovery| - [MetricsConfigHelper] defaultDiscovery - [LOCAL] [dev] [5.0.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
22:50:39,888  INFO |defaultDiscovery| - [logo] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
22:50:39,888  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
22:50:39,888  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Hazelcast Platform 5.0.1-SNAPSHOT (20211011 - 008f2fd) starting at [127.0.0.1]:5701
22:50:39,888  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Cluster name: dev
22:50:39,888  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, please do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true);
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true
  - Add environment variable: HZ_JET_ENABLED=true
22:50:39,891  INFO |defaultDiscovery| - [MetricsConfigHelper] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
22:50:39,895  INFO |defaultDiscovery| - [Node] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Using Multicast discovery
22:50:39,895  WARN |defaultDiscovery| - [CPSubsystem] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
22:50:39,899  INFO |defaultDiscovery| - [Diagnostics] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
22:50:39,899  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is STARTING
22:50:42,068  INFO |defaultDiscovery| - [ClusterService] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b this
]

22:50:42,068  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is STARTED
22:50:42,069  INFO |defaultDiscovery| - [MetricsConfigHelper] defaultDiscovery - [LOCAL] [dev] [5.0.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
22:50:42,070  INFO |defaultDiscovery| - [logo] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
22:50:42,070  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
22:50:42,070  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Hazelcast Platform 5.0.1-SNAPSHOT (20211011 - 008f2fd) starting at [127.0.0.1]:5702
22:50:42,070  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Cluster name: dev
22:50:42,070  INFO |defaultDiscovery| - [system] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, please do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true);
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true
  - Add environment variable: HZ_JET_ENABLED=true
22:50:42,072  INFO |defaultDiscovery| - [MetricsConfigHelper] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
22:50:42,077  INFO |defaultDiscovery| - [Node] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Using Multicast discovery
22:50:42,077  WARN |defaultDiscovery| - [CPSubsystem] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
22:50:42,082  INFO |defaultDiscovery| - [Diagnostics] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
22:50:42,082  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is STARTING
22:50:42,198  INFO |defaultDiscovery| - [MulticastJoiner] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Trying to join to discovered node: [127.0.0.1]:5701
22:50:42,199  INFO |defaultDiscovery| - [TcpServerConnection] hz.gracious_elbakyan.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:40212
22:50:42,199  INFO |defaultDiscovery| - [TcpServerConnection] hz.vibrant_elbakyan.IO.thread-in-0 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:40212 and /127.0.0.1:5701
22:50:44,199  INFO |defaultDiscovery| - [ClusterService] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b this
	Member [127.0.0.1]:5702 - 07d51cb7-5acd-4e39-ae1e-d623fe5b541a
]

22:50:44,200  INFO |defaultDiscovery| - [ClusterService] hz.vibrant_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b
	Member [127.0.0.1]:5702 - 07d51cb7-5acd-4e39-ae1e-d623fe5b541a this
]

22:50:44,805  WARN |defaultDiscovery| - [NodeMulticastListener] hz.vibrant_elbakyan.MulticastThread - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] New join request has been received from current master. Suspecting [127.0.0.1]:5701
22:50:44,805  WARN |defaultDiscovery| - [MembershipManager] hz.vibrant_elbakyan.MulticastThread - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b is suspected to be dead for reason: New join request has been received from current master. Suspecting [127.0.0.1]:5701
22:50:44,805  INFO |defaultDiscovery| - [MembershipManager] hz.vibrant_elbakyan.MulticastThread - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Starting mastership claim process...
22:50:44,805  INFO |defaultDiscovery| - [MembershipManager] hz.vibrant_elbakyan.MulticastThread - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Local MembersView{version=2, members=[MemberInfo{address=[127.0.0.1]:5701, uuid=e0ff9abc-a9b5-420a-8b56-a69dab21430b, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5702, uuid=07d51cb7-5acd-4e39-ae1e-d623fe5b541a, liteMember=false, memberListJoinVersion=2}]} with suspected members: [Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b] and initial addresses to ask: []
22:50:44,805  INFO |defaultDiscovery| - [TcpServerConnection] hz.vibrant_elbakyan.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Connection[id=1, /0.0.0.0:40212->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
22:50:44,807  INFO |defaultDiscovery| - [TcpServerConnection] hz.gracious_elbakyan.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Connection[id=1, /127.0.0.1:5701->/127.0.0.1:40212, qualifier=null, endpoint=[127.0.0.1]:5702, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
22:50:44,807  INFO |defaultDiscovery| - [ClusterService] hz.vibrant_elbakyan.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5702 - 07d51cb7-5acd-4e39-ae1e-d623fe5b541a this
]

22:50:44,807  INFO |defaultDiscovery| - [MembershipManager] hz.vibrant_elbakyan.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Mastership is claimed with: MembersView{version=3, members=[MemberInfo{address=[127.0.0.1]:5702, uuid=07d51cb7-5acd-4e39-ae1e-d623fe5b541a, liteMember=false, memberListJoinVersion=2}]}
22:50:44,807  INFO |defaultDiscovery| - [TransactionManagerService] hz.vibrant_elbakyan.cached.thread-5 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5701, UUID: e0ff9abc-a9b5-420a-8b56-a69dab21430b
22:50:44,808  INFO |defaultDiscovery| - [TcpServerConnector] hz.gracious_elbakyan.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 10000, bind-any: true
22:50:44,808  INFO |defaultDiscovery| - [TcpServerConnection] hz.vibrant_elbakyan.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:32800
22:50:44,808  INFO |defaultDiscovery| - [TcpServerConnection] hz.gracious_elbakyan.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:32800 and /127.0.0.1:5702
22:50:44,899  INFO |defaultDiscovery| - [ExplicitSuspicionOp] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5702
22:50:44,899  INFO |defaultDiscovery| - [TcpServerConnection] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Connection[id=2, /0.0.0.0:32800->/127.0.0.1:5702, qualifier=null, endpoint=[127.0.0.1]:5702, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: explicit suspicion
22:50:44,899  INFO |defaultDiscovery| - [MembershipManager] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 07d51cb7-5acd-4e39-ae1e-d623fe5b541a
22:50:44,899  INFO |defaultDiscovery| - [TcpServerConnection] hz.vibrant_elbakyan.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Connection[id=2, /127.0.0.1:5702->/127.0.0.1:32800, qualifier=null, endpoint=[127.0.0.1]:5701, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
22:50:44,899  INFO |defaultDiscovery| - [ClusterService] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b this
]

22:50:44,899  INFO |defaultDiscovery| - [TransactionManagerService] hz.gracious_elbakyan.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 07d51cb7-5acd-4e39-ae1e-d623fe5b541a
22:50:45,058  INFO |defaultDiscovery| - [InternalPartitionService] hz.vibrant_elbakyan.migration - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: 0
22:50:45,058  INFO |defaultDiscovery| - [InternalPartitionService] hz.vibrant_elbakyan.migration - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Most recent partition table is determined.
22:50:45,199  INFO |defaultDiscovery| - [ClusterService] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5702 - 07d51cb7-5acd-4e39-ae1e-d623fe5b541a this
]

22:50:45,199  WARN |defaultDiscovery| - [Node] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Config seed port is 5701 and cluster size is 1. Some of the ports seem occupied!
22:50:45,199  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is STARTED
22:50:45,201  INFO |defaultDiscovery| - [AbstractConfigLocator] defaultDiscovery - Loading 'hazelcast-client-default.xml' from the classpath.
22:50:45,215  INFO |defaultDiscovery| - [ClientInvocationService] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] Running with 2 response threads, dynamic=true
22:50:45,216  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211011 - 008f2fd) is STARTING
22:50:45,216  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211011 - 008f2fd) is STARTED
22:50:45,217  INFO |defaultDiscovery| - [ClientConnectionManager] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] Trying to connect to cluster: dev
22:50:45,217  INFO |defaultDiscovery| - [ClientConnectionManager] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] Trying to connect to [127.0.0.1]:5701
22:50:45,219 DEBUG |defaultDiscovery| - [AuthenticationMessageTask] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Processing authentication with clientUuid 7e01b51a-1e35-4d55-b802-4711ee08fbe5 and clientName hz.client_765
22:50:45,219  INFO |defaultDiscovery| - [AuthenticationMessageTask] hz.gracious_elbakyan.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Received auth from Connection[id=3, /127.0.0.1:5701->/127.0.0.1:38641, qualifier=null, endpoint=[127.0.0.1]:38641, alive=true, connectionType=JVM, planeIndex=-1], successfully authenticated, clientUuid: 7e01b51a-1e35-4d55-b802-4711ee08fbe5, client name: hz.client_765, client version: 5.0.1-SNAPSHOT
22:50:45,219  INFO |defaultDiscovery| - [LifecycleService] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211011 - 008f2fd) is CLIENT_CONNECTED
22:50:45,219  INFO |defaultDiscovery| - [ClientConnectionManager] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] Authenticated with server [127.0.0.1]:5701:e0ff9abc-a9b5-420a-8b56-a69dab21430b, server version: 5.0.1-SNAPSHOT, local address: /127.0.0.1:38641
22:50:45,219  INFO |defaultDiscovery| - [Diagnostics] defaultDiscovery - hz.client_765 [dev] [5.0.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
22:50:45,219 TRACE |defaultDiscovery| - [ClientListenerService] hz.client_765.internal-1 - hz.client_765 [dev] [5.0.1-SNAPSHOT] Register attempt of ClusterViewListenerHandler to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:38641->/127.0.0.1:5701}, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-10-11 22:50:45.219, lastWriteTime=2021-10-11 22:50:45.218, closedTime=never, connected server version=5.0.1-SNAPSHOT}
22:50:45,283 TRACE |defaultDiscovery| - [ClientListenerService] ForkJoinPool.commonPool-worker-3 - hz.client_765 [dev] [5.0.1-SNAPSHOT] Registered ClusterViewListenerHandler to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:38641->/127.0.0.1:5701}, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-10-11 22:50:45.220, lastWriteTime=2021-10-11 22:50:45.219, closedTime=never, connected server version=5.0.1-SNAPSHOT}
22:50:45,283  INFO |defaultDiscovery| - [ClientClusterService] hz.client_765.event-24071 - hz.client_765 [dev] [5.0.1-SNAPSHOT] 

Members [1] {
	Member [127.0.0.1]:5701 - e0ff9abc-a9b5-420a-8b56-a69dab21430b
}

22:50:45,284 TRACE |defaultDiscovery| - [ClientListenerService] hz.client_765.eventRegistration- - hz.client_765 [dev] [5.0.1-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@ccc12c4, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@3a4adb32} to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:38641->/127.0.0.1:5701}, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-10-11 22:50:45.220, lastWriteTime=2021-10-11 22:50:45.219, closedTime=never, connected server version=5.0.1-SNAPSHOT}
22:50:45,285 TRACE |defaultDiscovery| - [ClientListenerService] hz.client_765.eventRegistration- - hz.client_765 [dev] [5.0.1-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@ccc12c4, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@3a4adb32} to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:38641->/127.0.0.1:5701}, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-10-11 22:50:45.285, lastWriteTime=2021-10-11 22:50:45.284, closedTime=never, connected server version=5.0.1-SNAPSHOT}
22:50:45,285  INFO |defaultDiscovery| - [ClientStatisticsService] defaultDiscovery - Client statistics is enabled with period 5 seconds.
22:52:45,688  INFO |defaultDiscovery| - [LifecycleService] main - hz.client_765 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211011 - 008f2fd) is SHUTTING_DOWN
22:52:45,689  INFO |defaultDiscovery| - [ClientConnectionManager] main - hz.client_765 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701:e0ff9abc-a9b5-420a-8b56-a69dab21430b, connection: ClientConnection{alive=false, connectionId=1, channel=NioChannel{/127.0.0.1:38641->/127.0.0.1:5701}, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-10-11 22:52:45.288, lastWriteTime=2021-10-11 22:52:45.288, closedTime=2021-10-11 22:52:45.689, connected server version=5.0.1-SNAPSHOT}
22:52:45,689  INFO |defaultDiscovery| - [LifecycleService] main - hz.client_765 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211011 - 008f2fd) is CLIENT_DISCONNECTED
22:52:45,689  INFO |defaultDiscovery| - [TcpServerConnection] hz.gracious_elbakyan.IO.thread-in-2 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Connection[id=3, /127.0.0.1:5701->/127.0.0.1:38641, qualifier=null, endpoint=[127.0.0.1]:38641, alive=false, connectionType=JVM, planeIndex=-1] closed. Reason: Connection closed by the other side
22:52:45,690  INFO |defaultDiscovery| - [ClientEndpointManager] hz.gracious_elbakyan.event-24062 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Destroying ClientEndpoint{connection=Connection[id=3, /127.0.0.1:5701->/127.0.0.1:38641, qualifier=null, endpoint=[127.0.0.1]:38641, alive=false, connectionType=JVM, planeIndex=-1], clientUuid=7e01b51a-1e35-4d55-b802-4711ee08fbe5, clientName=hz.client_765, authenticated=true, clientVersion=5.0.1-SNAPSHOT, creationTime=1633992645218, latest clientAttributes=lastStatisticsCollectionTime=1633992765285,enterprise=false,clientType=JVM,clientVersion=5.0.1-SNAPSHOT,clusterConnectionTimestamp=1633992645218,clientAddress=127.0.0.1,clientName=hz.client_765,credentials.principal=null,os.committedVirtualMemorySize=27987353600,os.freePhysicalMemorySize=712961757184,os.freeSwapSpaceSize=4294963200,os.maxFileDescriptorCount=1048576,os.openFileDescriptorCount=367,os.processCpuTime=1627110000000,os.systemLoadAverage=4.55,os.totalPhysicalMemorySize=811309408256,os.totalSwapSpaceSize=4294963200,runtime.availableProcessors=8,runtime.freeMemory=3784322152,runtime.maxMemory=4294967296,runtime.totalMemory=4294967296,runtime.uptime=2907645,runtime.usedMemory=510645144, labels=[]}
22:52:45,690  INFO |defaultDiscovery| - [LifecycleService] main - hz.client_765 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211011 - 008f2fd) is SHUTDOWN
22:52:45,691  INFO |defaultDiscovery| - [LifecycleService] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
22:52:45,691  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutting down multicast service...
22:52:45,691  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutting down connection manager...
22:52:45,692  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutting down node engine...
22:52:45,695  INFO |defaultDiscovery| - [NodeExtension] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Destroying node NodeExtension.
22:52:45,695  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Hazelcast Shutdown is completed in 4 ms.
22:52:45,695  INFO |defaultDiscovery| - [LifecycleService] main - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
22:52:45,695  INFO |defaultDiscovery| - [LifecycleService] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
22:52:45,696  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Shutting down multicast service...
22:52:45,696  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Shutting down connection manager...
22:52:45,698  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Shutting down node engine...
22:52:45,700  INFO |defaultDiscovery| - [NodeExtension] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Destroying node NodeExtension.
22:52:45,701  INFO |defaultDiscovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
22:52:45,701  INFO |defaultDiscovery| - [LifecycleService] main - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
BuildInfo right after defaultDiscovery(com.hazelcast.client.impl.spi.impl.discovery.ClientAutoDetectionDiscoveryTest): BuildInfo{version='5.0.1-SNAPSHOT', build='20211011', buildNumber=20211011, revision=008f2fd, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'defaultDiscovery(com.hazelcast.client.impl.spi.impl.discovery.ClientAutoDetectionDiscoveryTest):'
22:50:35, accumulated pauses: 39 ms, max pause: 3 ms, pauses over 1000 ms: 0
22:50:40, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:50:45, accumulated pauses: 89 ms, max pause: 54 ms, pauses over 1000 ms: 0
22:50:50, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:50:55, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:00, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:05, accumulated pauses: 34 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:10, accumulated pauses: 33 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:15, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:20, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:25, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:30, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:35, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:40, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:45, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:51:50, accumulated pauses: 37 ms, max pause: 1 ms, pauses over 1000 ms: 0
22:51:55, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:00, accumulated pauses: 38 ms, max pause: 2 ms, pauses over 1000 ms: 0
22:52:05, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:10, accumulated pauses: 37 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:15, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:20, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:25, accumulated pauses: 35 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:30, accumulated pauses: 36 ms, max pause: 0 ms, pauses over 1000 ms: 0
22:52:35, accumulated pauses: 51 ms, max pause: 5 ms, pauses over 1000 ms: 0
22:52:40, accumulated pauses: 64 ms, max pause: 1 ms, pauses over 1000 ms: 0
22:52:45, accumulated pauses: 257 ms, max pause: 253 ms, pauses over 1000 ms: 0
```

--- a/hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/NodeMulticastListener.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/NodeMulticastListener.java
@@ -20,6 +20,9 @@ import com.hazelcast.internal.cluster.Joiner;
 import com.hazelcast.cluster.impl.MemberImpl;
 import com.hazelcast.instance.impl.Node;
 import com.hazelcast.logging.ILogger;
+
+import java.util.UUID;
+
 import com.hazelcast.cluster.Address;
 import com.hazelcast.internal.nio.Packet;
 
@@ -70,11 +73,17 @@ public class NodeMulticastListener implements MulticastListener {
             node.multicastService.send(response);
         } else if (joinMessage.getAddress().equals(masterAddress)) {
             MemberImpl master = node.getClusterService().getMember(masterAddress);
-            if (master != null && !master.getUuid().equals(joinMessage.getUuid())) {
-                String message = "New join request has been received from current master. Suspecting " + masterAddress;
-                logger.warning(message);
-                // I just make a local suspicion. Probably other nodes will eventually suspect as well.
-                clusterService.suspectMember(master, message, false);
+            if (master != null) {
+                UUID uuidFromMaster = master.getUuid();
+                UUID uuidInJoinRequest = joinMessage.getUuid();
+                if (!uuidFromMaster.equals(uuidInJoinRequest)) {
+                    String message = "New join request has been received from current master address. "
+                            + "The UUID in the join request (" + uuidInJoinRequest + ") is different from the "
+                            + "known master one (" + uuidFromMaster + "). Suspecting the master address: " + masterAddress;
+                    logger.warning(message);
+                    // I just make a local suspicion. Probably other nodes will eventually suspect as well.
+                    clusterService.suspectMember(master, message, false);
+                }
             }
         }
     }

--- a/hazelcast/src/test/java/com/hazelcast/client/cache/ClientCacheConfigTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/cache/ClientCacheConfigTest.java
@@ -46,6 +46,7 @@ import java.net.URL;
 import java.util.Properties;
 
 import static com.hazelcast.cache.CacheTestSupport.createClientCachingProvider;
+import static com.hazelcast.test.AbstractHazelcastClassRunner.getTestMethodName;
 import static com.hazelcast.test.Accessors.getNodeEngineImpl;
 import static com.hazelcast.test.TestEnvironment.isSolaris;
 import static org.junit.Assert.assertEquals;
@@ -172,7 +173,7 @@ public class ClientCacheConfigTest extends HazelcastTestSupport {
         HazelcastInstance server2 = null;
 
         try {
-            Config config = new Config();
+            Config config = new Config().setClusterName(getTestMethodName());
             if (isSolaris()) {
                 config.setProperty(ClusterProperty.MULTICAST_SOCKET_SET_INTERFACE.getName(), "false");
             }
@@ -189,7 +190,8 @@ public class ClientCacheConfigTest extends HazelcastTestSupport {
             ICacheService cacheService2 = getCacheService(server2);
 
             // Create the hazelcast client instance
-            client = HazelcastClient.newHazelcastClient();
+            ClientConfig clientConfig = new ClientConfig().setClusterName(getTestMethodName());
+            client = HazelcastClient.newHazelcastClient(clientConfig);
 
             // Create the client cache manager
             CachingProvider cachingProvider = createClientCachingProvider(client);

--- a/hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/discovery/ClientAutoDetectionDiscoveryTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/impl/spi/impl/discovery/ClientAutoDetectionDiscoveryTest.java
@@ -30,6 +30,7 @@ import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import static com.hazelcast.test.AbstractHazelcastClassRunner.getTestMethodName;
 import static com.hazelcast.test.TestEnvironment.isSolaris;
 
 @RunWith(HazelcastSerialClassRunner.class)
@@ -46,6 +47,8 @@ public class ClientAutoDetectionDiscoveryTest extends HazelcastTestSupport {
     @Test
     public void defaultDiscovery() {
         Config c = new Config();
+        c.setClusterName(getTestMethodName());
+
         if (isSolaris()) {
             c.setProperty(ClusterProperty.MULTICAST_SOCKET_SET_INTERFACE.getName(), "false");
         }
@@ -53,7 +56,8 @@ public class ClientAutoDetectionDiscoveryTest extends HazelcastTestSupport {
         Hazelcast.newHazelcastInstance(c);
         Hazelcast.newHazelcastInstance(c);
 
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        ClientConfig clientConfig = new ClientConfig().setClusterName(getTestMethodName());
+        HazelcastInstance client = HazelcastClient.newHazelcastClient(clientConfig);
         assertClusterSizeEventually(2, client);
     }
 

